src/event.py

```diff
@@ -0,0 +1,165 @@
+        vehicle_label,
+        updated_at,
+    ) = reduce_update_event(update)
+
+    prev = current_stop_state.get(
+        trip_id,
+        {
+            "stop_sequence": current_stop_sequence,
+            "stop_id": stop_id,
+            "updated_at": updated_at,
+            "event_type": event_type,
+        },
+    )
+
+    # current_stop_state updated_at is isofmt str, not datetime.
+    if isinstance(prev["updated_at"], str):
+        prev["updated_at"] = datetime.fromisoformat(prev["updated_at"])
+
+    is_departure_event, is_arrival_event = arr_or_dep_event(
+        prev=prev,
+        current_status=current_status,
+        current_stop_sequence=current_stop_sequence,
+        event_type=event_type,
+        stop_id=stop_id,
+    )
+
+    if is_departure_event or is_arrival_event:
+        if is_departure_event:
+            stop_id = prev["stop_id"]
+
+        stop_name = get_stop_name(stops, stop_id)
+        service_date = util.service_date(updated_at)
+
+        # refresh the gtfs data bundle if the day has incremented
+        if gtfs_service_date != service_date:
+            logger.info(f"Refreshing GTFS bundle from {gtfs_service_date} to {service_date}...")
+            gtfs_service_date = service_date
+            scheduled_trips, scheduled_stop_times, stops = gtfs.read_gtfs(gtfs_service_date)
+
+        if stop_id in STOPS.get(route_id, {}):
+            logger.info(
+                f"[{updated_at.isoformat()}] Event: route={route_id} trip_id={trip_id} {event_type} stop={stop_name}"
+            )
+
+            # write the event here
+            df = pd.DataFrame(
+                [
+                    {
+                        "service_date": service_date,
+                        "route_id": route_id,
+                        "trip_id": trip_id,
+                        "direction_id": direction_id,
+                        "stop_id": stop_id,
+                        "stop_sequence": current_stop_sequence,
+                        "vehicle_id": "0",  # TODO??
+                        "vehicle_label": vehicle_label,
+                        "event_type": event_type,
+                        "event_time": updated_at,
+                    }
+                ],
+                index=[0],
+            )
+
+            event = enrich_event(df, scheduled_trips, scheduled_stop_times)
+            disk.write_event(event)
+
+    current_stop_state[trip_id] = {
+        "stop_sequence": current_stop_sequence,
+        "stop_id": stop_id,
+        "updated_at": updated_at,
+        "event_type": event_type,
+    }
+
+    # write the state out here
+    disk.write_state(current_stop_state)
+
+
+def enrich_event(df: pd.DataFrame, scheduled_trips: pd.DataFrame, scheduled_stop_times: pd.DataFrame):
+    """
+    Given a dataframe with a single event, enrich it with headway information and return a single event dict
+    """
+    headway_adjusted_df = gtfs.add_gtfs_headways(df, scheduled_trips, scheduled_stop_times)
```

Comment:
ty for this! we'll want to put the explicit tz conversion of `event_time` to eastern before we add GTFS headways, because the scheduled stop times we are merging against there are provided in eastern

Reply:
Got it, moved above